src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -2847,13 +2717,13 @@ public void emitPrivateValue(String emitName) {
 
     public Void emitObjectType(
         ObjectType type, boolean extendingInstanceClass, boolean inExtendsImplementsPosition) {
-      if (maybeEmitGlobalAlias(type)) return null;
       // Closure doesn't require that all the type params be declared, but TS does
       if (!type.getTemplateTypeMap().isEmpty()
           && !typeRegistry.getNativeType(OBJECT_TYPE).equals(type)) {
         return emitTemplatizedType(
             typeRegistry.createTemplatizedType(type), false, inExtendsImplementsPosition);
       }
+      if (maybeEmitGlobalType(type)) return null;
```

Comment:
odd that this had to move

Reply:
Without it some tests fail because we write Array in place of any[].